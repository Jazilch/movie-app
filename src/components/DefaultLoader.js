'use es6';

import React from 'react';
import styled from 'styled-components';
import { css } from '@emotion/core';
import ScaleLoader from 'react-spinners/ScaleLoader';

const LoadingGrid = styled.section`
  display: flex;
  height: 100%;
  justify-content: center;
  align-items: center;
`;

const override = css`
  display: block;
  margin: 0 auto;
  text-align: center;
`;

const DefaultLoader = () => {
  return (
    <LoadingGrid>
      <ScaleLoader css={override} sizeUnit="px" size={150} color="red" />
    </LoadingGrid>
  );
};

export default DefaultLoader;
