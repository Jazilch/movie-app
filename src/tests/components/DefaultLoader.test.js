'use es6';

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DefaultLoader from '../../components/DefaultLoader';

describe('Default Loader component', () => {
  it('renders the component correctly', () => {
    const defaultloader = shallow(<DefaultLoader />);
    expect(toJson(defaultloader)).toMatchSnapshot();
  });
});
