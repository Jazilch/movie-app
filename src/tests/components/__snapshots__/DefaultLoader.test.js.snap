// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Loader component renders the component correctly 1`] = `
<styled.section>
  <Loader
    color="red"
    css={
      Object {
        "map": undefined,
        "name": "f37jqc",
        "next": undefined,
        "styles": "
  display: block;
  margin: 0 auto;
  text-align: center;
",
        "toString": [Function],
      }
    }
    height={35}
    heightUnit="px"
    loading={true}
    margin="2px"
    radius={2}
    radiusUnit="px"
    size={150}
    sizeUnit="px"
    width={4}
    widthUnit="px"
  />
</styled.section>
`;
